File: Challenges/September 2019 Challenge/Easyfib.cpp
#include<iostream>
#include<vector>

using namespace std;

vector<int> fib(int n){
    vector<int>fibResult;
    fibResult.push_back(0);
    fibResult.push_back(1);

    for(int i=2;i<n;i++){
        fibResult.push_back(fibResult[i-1] + fibResult[i-2]);
    }

    return fibResult;
}

void newFibValue(vector<int> arr){
    vector<int> newArray;

    for(int i=0;i<arr.size();i++){
        if((i+1)%2 == 0){
            newArray.push_back(arr[i]%10);
        }
    }

    if(newArray.size() == 1)
        cout<<newArray[0]<<endl;
    else{
        arr.clear();
        arr = newArray;
        newFibValue(arr);
    }

}
int main(){
    int t;
    cin>>t;

    while(t)
    {
        int n;
        cin>>n;
        vector<int> fibResult = fib(n);
        newFibValue(fibResult);
        //cout<<result<<endl;
        t = t - 1;
    }
    
    return 0;
}

File: Challenges/September 2019 Challenge/Easyfib.py
import sys

testCases = sys.stdin.readline()
testCases = int(testCases)

def fib(n):
    Fibarr = list()
    Fibarr.append(0)
    Fibarr.append(1)

    for i in range(2,n):
        Fibarr.append(Fibarr[i-1] + Fibarr[i-2])
    
    return Fibarr

def easyFibvalue(fibresult):
    newList = list()
    for i in range(0,len(fibresult)):
        if((i+1)%2 == 0):
            newList.append(fibresult[i])
    
    #print(newList)
    if(len(newList) == 1):
        print(newList[-1])
    else:
        fibresult = newList
        easyFibvalue(fibresult)
        newList.clear()
        del(newList)

while testCases != 0:
    n = sys.stdin.readline()
    n = int(n)
    FibonacciResult = fib(n)
   # print(FibonacciResult)
    easyFibvalue([x%10 for x in FibonacciResult])
    testCases -= 1


